e2e/tests-dfx/dotenv.bash

```diff
@@ -51,15 +51,48 @@ teardown() {
   assert_match "The output_env_file must be within the project root, but is .*/working-dir/e2e_project/../outside/.env"
 }
 
-@test "writes environment variables to .env" {
+@test "build writes all environment variables to .env" {
   dfx_start
   dfx canister create --all
+
+  writes_all_environment_variables_to_env build
+}
+
+@test "deploy writes all environment variables to .env" {
+  dfx_start
+  dfx canister create --all
+
+  writes_all_environment_variables_to_env deploy
+}
+
+@test "canister install writes all environment variables to .env" {
+  dfx_start
+  dfx canister create --all
+  dfx build e2e_project_frontend
+  # set a post-install script so the install will create a .env file
+  jq '.canisters.e2e_project_frontend.post_install="echo post install"' dfx.json | sponge dfx.json
+
+  writes_all_environment_variables_to_env canister install
+}
+
+writes_all_environment_variables_to_env() {
```

Comment:
```suggestion
assert_dotenv_contains_all_variables_after_command() {
```

I was confused when I see no assertions in the new tests above.
Then I found out that this method make assertions inside.